tests/ui/abi/riscv32e-registers.rs

```diff
@@ -0,0 +1,92 @@
+// Test that loads into registers x16..=x31 are never generated for riscv32{e,em,emc} targets
+//
+//@ build-fail
+//@ revisions: riscv32e riscv32em riscv32emc
+//
+//@ compile-flags: --crate-type=rlib
+//@ [riscv32e] needs-llvm-components: riscv
+//@ [riscv32e] compile-flags: --target=riscv32e-unknown-none-elf
+//@ [riscv32em] needs-llvm-components: riscv
+//@ [riscv32em] compile-flags: --target=riscv32em-unknown-none-elf
+//@ [riscv32emc] needs-llvm-components: riscv
+//@ [riscv32emc] compile-flags: --target=riscv32emc-unknown-none-elf
+
+#![no_core]
+#![feature(no_core, lang_items, rustc_attrs)]
+
+#[rustc_builtin_macro]
+macro_rules! asm {
+    () => {};
+}
+
+#[lang = "sized"]
+trait Sized {}
+
+// Check that loads to registers x1..=x16 will be generated but loads to registers x17..=x31 will
+// not.
```

Comment:
typo
```suggestion
// Check that loads to registers x1..=x15 will be generated but loads to registers x16..=x31 will
// not.
```

or perhaps
```suggestion
// Verify registers x1..=x15 are addressable on riscv32e, but registers x16..=x31 are not.
```